File: framework/languages/dotnet/samples/Bingo(session-gateway)/Server/Api/AuthenticatePlayerHandler.cs
using Bingo.SessionGateway.Shared.Contracts;

namespace Bingo.SessionGateway.Api;

internal sealed class AuthenticatePlayerHandler
    : IZLinkRequestHandler<AuthenticatePlayerReq, AuthenticatePlayerRes>
{
    public ValueTask<AuthenticatePlayerRes> HandleAsync(
        AuthenticatePlayerReq request,
        ZLinkRequestContext context,
        CancellationToken cancellationToken)
    {
        cancellationToken.ThrowIfCancellationRequested();
        if (!request.AccessToken.StartsWith("player-", StringComparison.Ordinal))
        {
            return ValueTask.FromResult(new AuthenticatePlayerRes(
                false,
                null,
                null,
                "Access token must be a sample player id."));
        }

        var displayName = request.AccessToken.Replace("player-", "Player ", StringComparison.Ordinal);
        return ValueTask.FromResult(new AuthenticatePlayerRes(
            true,
            request.AccessToken,
            displayName,
            null));
    }
}

File: framework/languages/node/samples/Bingo.Ts/Server/Api/Handlers/authenticate-player-handler.ts
import { PacketNames, authenticatePlayerAccepted, authenticatePlayerRejected } from '../../../Shared/Contracts/messages';
import { SampleNames } from '../../Configuration/sample-names';
import { zlinkRequestHandler } from '@zlink-systems/nestjs';
import type { ZLinkRequestHandler } from '@zlink-systems/framework';
import type {
  AuthenticatePlayerRes,
  AuthenticatePlayerReq
} from '../../../Shared/Contracts/messages';

@zlinkRequestHandler('api', PacketNames.authenticatePlayerReq)
class AuthenticatePlayerHandler implements ZLinkRequestHandler<AuthenticatePlayerReq, AuthenticatePlayerRes> {
  async handle(request: AuthenticatePlayerReq): Promise<AuthenticatePlayerRes> {
    if (!(SampleNames.actorIds as readonly string[]).includes(request.accessToken)) {
      return authenticatePlayerRejected('Access token must be a sample player id.');
    }

    return authenticatePlayerAccepted(request.accessToken);
  }
}

export { AuthenticatePlayerHandler };
